Source/Diagnostics/ReducedDiags/FieldReduced.cpp

```diff
@@ -0,0 +1,292 @@
+                    ofs << "max_|E|_lev"+std::to_string(lev)+" (V/m)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+
+                                                shift_Bx+noutputs_maxField*lev) + "]";
+                    ofs << "max_Bx_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+
+                                                shift_By+noutputs_maxField*lev) + "]";
+                    ofs << "max_By_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+
+                                                shift_Bz+noutputs_maxField*lev) + "]";
+                    ofs << "max_Bz_lev"+std::to_string(lev)+" (T)";
+                    ofs << m_sep;
+                    ofs << "[" + std::to_string(m_offset_maxField+
+                                                shift_absB+noutputs_maxField*lev) + "]";
+                    ofs << "max_|B|_lev"+std::to_string(lev)+" (T)";
+                }
+            }
+
+            ofs << std::endl;
+            // close file
+            ofs.close();
+        }
+    }
+
+}
+// end constructor
+
+// function that computes field energy and/or maximum field values
+void FieldReduced::ComputeDiags (int step)
+{
+
+    // Judge if the diags should be done
+    if ( (step+1) % m_freq != 0 ) { return; }
+
+    // get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    // get number of level
+    const auto nLevel = warpx.finestLevel() + 1;
+
+    // loop over refinement levels
+    for (int lev = 0; lev < nLevel; ++lev)
+    {
+
+        // get MultiFab data at lev
+        const MultiFab & Ex = warpx.getEfield(lev,0);
+        const MultiFab & Ey = warpx.getEfield(lev,1);
+        const MultiFab & Ez = warpx.getEfield(lev,2);
+        const MultiFab & Bx = warpx.getBfield(lev,0);
+        const MultiFab & By = warpx.getBfield(lev,1);
+        const MultiFab & Bz = warpx.getBfield(lev,2);
+
+        if (m_fieldEnergy)
+        {
+            // get cell size
+            Geometry const & geom = warpx.Geom(lev);
+            auto domain_box = geom.Domain();
+#if (AMREX_SPACEDIM == 2)
+            auto dV = geom.CellSize(0) * geom.CellSize(1);
+#elif (AMREX_SPACEDIM == 3)
+            auto dV = geom.CellSize(0) * geom.CellSize(1) * geom.CellSize(2);
+#endif
+
+            // compute E squared
+            Real tmpx = Ex.norm2(0,geom.periodicity());
+            Real tmpy = Ey.norm2(0,geom.periodicity());
+            Real tmpz = Ez.norm2(0,geom.periodicity());
+            Real Es = tmpx*tmpx + tmpy*tmpy + tmpz*tmpz;
+
+            // compute B squared
+            tmpx = Bx.norm2(0,geom.periodicity());
+            tmpy = By.norm2(0,geom.periodicity());
+            tmpz = Bz.norm2(0,geom.periodicity());
+            Real Bs = tmpx*tmpx + tmpy*tmpy + tmpz*tmpz;
```

Comment:
No need to re-use var names. This is more explicit about the temporary nature of the vars:
```suggestion
            Real const tmpEx = Ex.norm2(0,geom.periodicity());
            Real const tmpEy = Ey.norm2(0,geom.periodicity());
            Real const tmpEz = Ez.norm2(0,geom.periodicity());
            Real const Es = tmpEx*tmpEx + tmpEy*tmpEy + tmpEz*tmpEz;

            // compute B squared
            Real const tmpBx = Bx.norm2(0,geom.periodicity());
            Real const tmpBx = By.norm2(0,geom.periodicity());
            Real const tmpBx = Bz.norm2(0,geom.periodicity());
            Real const Bs = tmpBx*tmpBx + tmpBy*tmpBy + tmpBz*tmpBz;
```